dev/io.openliberty.microprofile.telemetry.2.0.logging.internal_fat/fat/src/io/openliberty/microprofile/telemetry/logging/internal_fat/TelemetryAgentJULMessagesTest.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.logging.internal_fat;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+@RunWith(FATRunner.class)
+public class TelemetryAgentJULMessagesTest extends FATServletClient {
+
+    private static Class<?> c = TelemetryAgentJULMessagesTest.class;
+
+    public static final String SERVER_NAME = "TelemetryAgentJULMessages";
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void initialSetup() throws Exception {
+        // Copy the OTel Java agent to the Liberty server root directory.
+        server.copyFileToLibertyServerRoot("agent-260/opentelemetry-javaagent.jar");
+        ShrinkHelper.defaultDropinApp(server, "ffdc-servlet", "io.openliberty.microprofile.telemetry.logging.internal.fat.ffdc.servlet");
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        if (server != null && !server.isStarted()) {
+            server.startServer();
+        }
+    }
+
+    @After
+    public void cleanUp() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    /**
+     * Ensure JUL message logs are not duplicated when the OpenTelemetry Java agent is active with mpTelemetry-2.0.
+     */
+    @Test
+    public void testNoDuplicateJULMessageLogsWithOpenTelemetryAgent() throws Exception {
+        // Wait for the JUL message (SRVE0250I) to appear in the logs.
+        server.waitForStringInLog("SRVE0250I");
+        List<String> agentMappedJulMsg = server.findStringsInLogs("SRVE0250I", server.getConsoleLogFile());
+
+        Log.info(c, "testNoDuplicateJULMessageLogsWithOpenTelemetryAgent()", "Found JUL messages: " + agentMappedJulMsg);
+
+        // There should only be one instance of the JUL message routed to OpenTelemetry.
+        assertTrue("There are duplicate JUL messages.", agentMappedJulMsg.size() == 1);
```

Comment:
On a second thought, this test may not be reliable.  The test waits for the first occurrence of `SRVE0250I` and find all occurrences from the console file.  At that time, the second (or the duplicated) message may not arrive yet, but the test will still pass even there are duplicates.

Reply:
@fmhwong Good point! My train of thought was that since the handler is synchronized, the msg will always be sent to `synchronizedWrite()` first, before the agent gets it. However, I see your point, sometimes the test maybe slow.

Do you think if we add a sleep for 15-30 seconds, and then find all the occurrences would resolve this?

Reply:
If the messages are arriving in order, we can wait for the 2nd message to arrive and check how many copies of the first message.  We don't need to waste 15-30 seconds if this works.

Reply:
@fmhwong Added a wait for a second message ID, since it appears right after the first message ID.